.github/workflows/validate-theme-entry.yml

```diff
@@ -0,0 +1,263 @@
+      - run: npm install probe-image-size
+      - uses: actions/github-script@v6
+        with:
+          script: |
+            const probe = require('probe-image-size');
+            const fs = require('fs');
+
+            if (context.payload.pull_request.additions <= context.payload.pull_request.deletions) {
+                // Don't run any validation checks if the user is just modifying existing theme config
+                return;
+            }
+
+            const errors = [];
+            const addError = (error) => {
+                errors.push(`:x: ${error}`);
+                console.log('Found issue: ' + error);
+            };
+
+            const warnings = [];
+            const addWarning = (warning) => {
+                warnings.push(`:warning: ${warning}`);
+                console.log('Found issue: ' + warning);
+            }
+
+            // core validation logic
+            await (async () => {
+                if (context.payload.pull_request.changed_files > 1) {
+                    addError('You modified files other than `community-css-themes.json`.');
+                }
+
+                let theme;
+
+                try {
+                    const themes = JSON.parse(fs.readFileSync('community-css-themes.json', 'utf8'));
+                    theme = themes[themes.length - 1];
+
+                } catch (e) {
+                    addError('Could not parse `community-css-themes.json`, invalid JSON. ' + e.message);
+                    return;
+                }
+
+                const author = context.payload.pull_request.user.login;
+                
+                let validPrKeys = ['name', 'author', 'repo', 'screenshot', 'modes'];
+                for (let key in Object.keys(manifest)) {
+                    if (!validPrKeys.includes(key)) {
+                        addError(`Your manifest has the invalid \`${key}\` property.`);
+                    }
+                }
+                for (let key in validPrKeys) {
+                    if (!manifest.hasOwnProperty(key)) {
+                        addError(`Your PR does not have the required \`${key}\` property.`);
+                    }
+                }
+
+                if (theme.name.toLowerCase().includes('obsidian')) {
+                    addWarning(`We discourage themes from including the word "Obsidian" in their name since it's redundant and makes the theme selection screen harder to visually parse.`);
+                }
+
+                if (themes.filter(t => t.name === theme.name).length > 1) {
+                    addError('There is already a theme with this name');
+                }
+
+                if (themes.filter(t => t.repo === theme.repo).length > 1) {
+                    addError('There is already a entry pointing to the `' + theme.repo + '` repository');
+                }
+
+                // we can't validate further, as there is no repo set
+                if(!theme.hasOwnProperty('repo')) {
```

Comment:
Shouldn't this check be moved a bit more earlier? We're already using `theme.repo` in the previous section.

Reply:
Actually no this is not needed since we already do a check for all the required keys.